orttraining/orttraining/core/graph/gradient_builder_base.h

```diff
@@ -173,24 +174,34 @@ class GradientBuilderBase {
                    {ONNX_NAMESPACE::MakeAttribute("value", t_proto)});
   }
 
-  static NodeDef ConstantValueNode(float value, const std::string& arg_name) {
+  static NodeDef ConstantValueNode(float value, const std::string& arg_name, int elem_type) {
     ONNX_NAMESPACE::TensorProto t_proto;
     t_proto.add_dims(1);
-    t_proto.set_data_type(ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
-    t_proto.add_float_data(value);
+    t_proto.set_data_type(elem_type);
+    if (elem_type == ONNX_NAMESPACE::TensorProto_DataType_FLOAT16) {
+      t_proto.add_int32_data(math::floatToHalf(value));
```

Comment:
>add_int32_data [](start = 14, length = 14)

I am not sure about this. 
Are float16 and bfloat16 both using the int field buffer? 

Or shall we use set_raw_data()?

Reply:
I borrowed the code from Cheng's branch. Struct BFloat16 and MLFloat16 are similar, both of them use uint16_t to store data internally, so reuse the INT field seems reasonable. @souptc do you know any reason to use INT field? Is there any standard from ONNX side?